### ISSUE TITLE
Save an email

### ISSUE THREAD

**Issue by WilliamMayor** [pazz/alot#1310]
**Is your feature request related to a problem? Please describe.**
I'd like to save emails as .eml files. When I get a receipt emailed to me I'd like to save it to my receipts directory for my accountant to deal with.

**Describe the solution you'd like**
A command for saving emails. Or pointers on how to write a custom keybinding that does this.

**Describe alternatives you've considered**
I've tried writing a custom keybind command that uses `pipeto` and combinations of `cat` or `tee` and regular redirects. I can't get it to save the email. Or at least I can, but not if I use spaces in the file name, and adding quotes doesn't work:

    pipeto "tee '/Users/william/Reciepts/DATE NAME.eml'"
    pipeto "cat > '/Users/william/Reciepts/DATE NAME.eml'"
    pipeto "cat" > '/Users/william/Reciepts/DATE NAME.eml'

**Additional context**
I'm very new to alot so I suspect I'm just missing the obvious way to do this. Any and all help appreciated. Thanks!

**Comment by pazz** [comment]
Hi and welcome!

Your use-case sounds as if it might be more helpful to just print the mail as pdf instead.
To do that, you can use muttprint: put the following in your `~/.config/alot/config`:

> print_cmd = muttprint -1 -F Utopia -a From_Date_*Subject* -p TO_FILE:~/print.pdf

then use the `print` command in thread buffers.

If you really wanted to write the email source to a file, there is no command for this as far as I remember.
But you could write your own python hook and call it using `call hookname` from within alot. 

With current alot master, your hook code would look similar to this:

```
async def write_mail(ui):
    # get msg content
    msg = ui.current_buffer.get_selected_message()  # this is an alot.db.Message
    eml = msg.get_email()  # this is an email.Message

    # open file and write str(eml)..
```

Check out the wiki for more inspiration on hooks.
I hope this helps and have fun hacking :)

**Comment by WilliamMayor** [comment]
Hi @pazz, thanks for your reply, and even more thanks for creating alot!

The PDF thing sounds good to me. Looks like I need to install mutt as well? Do I need some kind of HTML to PDF tool as well? Or does muttprint take care of that?

I'll explore the hook option as well.

Thanks!

**Comment by WilliamMayor** [comment]
Hey, I've just tried writing a custom hook. I've added some simple print statements to a function in `.config/alot/hooks.py`. I'm then trying to run that function by running `call FUNCTION` from the alot prompt. It gives me an error that 'name FUNCTION if not defined'.

I can't see anything in the docs about custom hooks like this. Is it a feature in a latest release? You mentioned being on the master, I'm actually on version 0.7.

Thanks for your help
